File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 1/13/2026, 10:49:40 PM
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size() ;
        int maxProfit=0;
        int minPrice= INT_MAX;
        for(int i=0; i<n; i++){
            if(prices[i]< minPrice){
                minPrice= prices[i];
            }
             if(prices[i]-minPrice > maxProfit){
                maxProfit= prices[i]-minPrice;
            }
        }
        return maxProfit;
       
    }

};

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.js
// Last updated: 1/13/2026, 10:49:47 PM
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(arr) {
    let maxP = 0;
    let minV = arr[0] 
    for(let i = 1 ; i < arr.length ; i++){
        if(arr[i] - minV > maxP){
            //update maxP
            maxP = arr[i]-minV;
            }
        if(arr[i] < minV){
            minV = arr[i];
        }
    }
    return maxP;
};